### ISSUE TITLE
unable to complete the training with contrastive loss only 

### ISSUE THREAD

**Issue by Horace89** [radekosmulski/whale#19]
Hi Radek,

I finally managed to train my network according to the first part of the "classification_and_metric_learning" notebook. 
I managed to train with SZ=224 and then proceeded with SZ=448 with the custom loss function. 

But so far, I've been unable to do the last part of the training successfully. The part where you mention: "For the last segment of the training, I used contrastive loss only." Indeed my network seems to not learn anything anymore.
Here is my code:
```
[...]
learn = Learner(create_fake_data(), CustomModel(), loss_func=contr_loss, metrics=[accuracy_mod, map5_mod])

learn.model_dir = Path(os.path.join(ROOT_DIR, 'data/models/'))
learn = learn.clip_grad()
learn.split((learn.model.cnn[6], learn.model.head))
learn.freeze()

learn.load('s2.3-var1a-oywrks063e'); #loaded from the training with SZ=448


dists = create_similarity_dict(learn.model, basic_dataloader)
learn.data = create_data(SZ, dists, BS)  

print('training started...', flush=True)
learn.fit_one_cycle(20, 1e-3)
```

But neither the validation loss nor the training loss is decreasing... The network does not learn anymore. 😕

Would you have any advice? Did you use a k smaller than 20, eg `learn.data = create_data(SZ, dists, BS, k=5)`? Do you remember the details of how you trained this last segment (contrastive loss only)?

**Comment by radekosmulski** [comment]
Very sorry @Horace89, it has been so long I don't remember.

**Comment by Horace89** [comment]
Too bad ;-) Thank you anyway!
